src/utils/string_utils.hpp

```diff
@@ -136,210 +136,66 @@ namespace StringUtils
-        return insertValues(s, all_vals);
-    }   // insertValues(s, v1, ..., v5)
-
-    // ------------------------------------------------------------------------
-    template <class T1, class T2, class T3, class T4>
-    std::string insertValues(const std::string &s, const T1 &v1,
-                             const T2 &v2, const T3 &v3, const T4 &v4)
-    {
-        std::vector<std::string> all_vals;
-        std::ostringstream dummy;
-        dummy << v1; all_vals.push_back(dummy.str()); dummy.str("");
-        dummy << v2; all_vals.push_back(dummy.str()); dummy.str("");
-        dummy << v3; all_vals.push_back(dummy.str()); dummy.str("");
-        dummy << v4; all_vals.push_back(dummy.str());
-        return insertValues(s, all_vals);
-    }   // insertValues(s, v1, ..., v4)
-
-    // ------------------------------------------------------------------------
-    /** Shortcut insert_values taking three values, see above for
-     *  full docs.
-     *  \param s String in which all %s or %d are replaced.
-     *  \param v1,v2, v3 Value(s) to replace all %s or %d with.
-     */
-    template <class T1, class T2, class T3>
-    std::string insertValues(const std::string &s, const T1 &v1,
-                             const T2 &v2, const T3 &v3)
-    {
-        std::vector<std::string> all_vals;
-        std::ostringstream dummy;
-        dummy << v1; all_vals.push_back(dummy.str()); dummy.str("");
-        dummy << v2; all_vals.push_back(dummy.str()); dummy.str("");
-        dummy << v3; all_vals.push_back(dummy.str());
-        return insertValues(s, all_vals);
-    }   // insertValues(s, v1, ..., v3)
-
-    // ------------------------------------------------------------------------
-    // Note: the order in which the templates are specified is important, since
-    // otherwise some compilers will not find the right template to use.
-    /** Shortcut insert_values taking three values, see above for
-     *  full docs.
-     *  \param s String in which all %s or %d are replaced.
-     *  \param v1,v2 Value(s) to replace all %s or %d with.
-     */
-    template <class T1, class T2>
-    std::string insertValues(const std::string &s, const T1 &v1,
-                             const T2 &v2)
-    {
-        std::vector<std::string> all_vals;
-        std::ostringstream dummy;
-        dummy << v1; all_vals.push_back(dummy.str()); dummy.str("");
-        dummy << v2; all_vals.push_back(dummy.str()); dummy.str("");
-
-        return insertValues(s, all_vals);
-    }   // insertValues(s, v1, v2)
-    // ------------------------------------------------------------------------
-    /** Shortcut insert_values taking three values, see above for
-     *  full docs.
-     *  \param s String in which all %s, %d are replaced.
-     *  \param v1 Value to replace.
-     */
-    template <class T1>
-    std::string insertValues(const std::string &s, const T1 &v1)
+    /** Intermediate struct to fill a vector using variadic templates */
```

Comment:
Can you explain more in detail what this does? Variadic templates are black magic ;) and having a nice long explanation will definitely help.
Also can we not use double underscore '__'? As far as I know the standard says that two underscore is reserved for the compilers internal use.